File: packages/shared/src/athlete-chat.ts
export const AI_CHAT_MAX_MESSAGE_LENGTH = 2000;

export interface AthleteAiChatThread {
  id: string;
  title: string | null;
  created_at: string;
  updated_at: string;
}

export interface AthleteAiChatMessage {
  id: string;
  role: "user" | "assistant" | string;
  content: string;
  created_at: string;
}

export interface AthleteAiChatSendResponse {
  user_message: AthleteAiChatMessage;
  assistant_message: AthleteAiChatMessage;
}

export interface AthleteAiChatThreadCreatePayload {
  title?: string | null;
}

export interface AthleteAiChatMessageCreatePayload {
  content: string;
}

File: backend/app/schemas/athlete_chat.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


MAX_CHAT_MESSAGE_LENGTH = 2000
MAX_CHAT_MESSAGES_PER_THREAD = 100
MAX_CHAT_HISTORY_FOR_LLM = 40


class AthleteAiChatThreadCreateRequest(BaseModel):
    title: str | None = Field(default=None, max_length=200)


class AthleteAiChatMessageCreateRequest(BaseModel):
    content: str = Field(min_length=1, max_length=MAX_CHAT_MESSAGE_LENGTH)


class AthleteAiChatThreadResponse(BaseModel):
    id: UUID
    title: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AthleteAiChatMessageResponse(BaseModel):
    id: UUID
    role: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AthleteAiChatSendResponse(BaseModel):
    user_message: AthleteAiChatMessageResponse
    assistant_message: AthleteAiChatMessageResponse
